beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -43,10 +43,39 @@ func (s *Service) roundRobinSync(genesis time.Time) error {
 	defer state.SkipSlotCache.Enable()
 
 	s.counter = ratecounter.NewRateCounter(counterSeconds * time.Second)
+
+	// Step 1 - Sync to end of finalized epoch.
+	if err := s.syncToFinalizedEpoch(ctx, genesis); err != nil {
+		return err
+	}
+
+	// Already at head, no need for 2nd phase.
+	if s.chain.HeadSlot() == helpers.SlotsSince(genesis) {
+		return nil
+	}
+
+	// Step 2 - sync to head from majority of peers (from no less than MinimumSyncPeers*2 peers)
+	// having the same world view on non-finalized epoch.
+	if err := s.syncToNonFinalizedEpoch(ctx, genesis); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// syncToFinalizedEpoch sync from head to best known finalized epoch.
+func (s *Service) syncToFinalizedEpoch(ctx context.Context, genesis time.Time) error {
 	highestFinalizedSlot, err := helpers.StartSlot(s.highestFinalizedEpoch() + 1)
 	if err != nil {
 		return err
 	}
+	// Set to the last slot of the finalized epoch.
+	highestFinalizedSlot--
```

Comment:
how do we handle missing slot in this case?

Reply:
It is not missing slot, we have a finalized epoch add +1 to it, and take the start slot of the **next to finalized epoch**. But since this method should sync to finalized slot only, we need to get the last slot of the finalized epoch, not the first slot of the next epoch.

Reply:
Although `highestFinalizedEpoch` cannot be zero (since we are using start slot of the next to finalized epoch), I still added check for zero, just in case some logic on how we first setting up the `highestFinalizedSlot` changes (and we will be open to underflow).

Reply:
thanks for clarifying